src/gleam/string.gleam

```diff
@@ -96,39 +96,94 @@ pub external fn compare(String, String) -> order.Order =
   "gleam_stdlib" "compare_strings"
 
 /// ## Get Substrings
-
-// TODO
-/// Take a substring given a start and end Grapheme indexes. Negative indexes
-/// are taken starting from the *end* of the list.
+///
+/// Take a substring given a starting index and length.
+/// A negative start or length will return an empty string.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
-/// slice("gleam", from: 1, to: 3) == "lea"
-/// slice("gleam", from: 1, to: 10) == "leam"
-/// slice("snakes on a plane!", from: -6, to: -1) == "plane"
+/// slice("gleam", from: 1, length: 3) == "lea"
+/// slice("gleam", from: 1, length: 10) == "leam"
+/// slice("snakes on a plane!", from: -6, to: -1) == ""
 /// ```
 ///
-// pub fn slice(out_of string: String, from start: Int, end: Int) -> String {}
+external fn erl_string_slice(string, start, length) -> String =
+  "string" "slice"
 
-// TODO
-/// Drop *n* Graphemes from the left side of a
+pub fn slice(out_of out_of: String, from from: Int, length length: Int) -> String {
+  let zero = 0
+
+  case from, length {
+    fr, _ if zero > fr ->
+      ""
+
+    _, ln if zero > ln ->
+      ""
+
+    fr, ln ->
+      erl_string_slice(out_of, fr, ln)
+  }
```

Comment:
I think this would be clearer and more concise like this:

```rust
case from < 0 || length <= 0 {
  True -> ""
  False -> erl_string_slice(out_of, fr, ln)
} 
```